doc/release/technotes/README.extern.rst

```diff
@@ -756,6 +764,167 @@ which demonstrates full integration between Chapel and the C function
+   address). That means that passing a c_ptr from one locale to another
+   and then using it on the second locale will probably result in a core
+   dump.
+
+Initialization
+--------------
+
+Chapel variables of extern type are not generally initialized
+automatically. Be sure to manually initialize Chapel variables of extern
+type.
+
+In the future, we would like to support automatic zero initialization
+of such variables and a way to provide their default initializer.
+
+Working with c_ptr
+------------------
+
+A c_ptr type can be constructed from a Chapel variable using the c_ptrTo()
+function; for example:
+
+.. code-block:: chapel
+
+ var i:c_int;
+ var i_ptr = c_ptrTo(i); // now i_ptr has type c_ptr(c_int) == int* in C
+
+Since a C pointer might refer to a single variable or an array, the c_ptr type
+supports 0-based array indexing and dereferencing. In addition, it is possible
+to allocate and free space for one or more elements and return the result as a
+c_ptr. See the following example:
+
+.. code-block:: chapel
+
+  var cArray = c_calloc(c_int, 10);
+  for i in 0..#10 {
+   cArray[i] = i:c_int;
+  }
+  // c_ptr.deref() always refers to the first element.
+  cArray.deref() = 17;
+  for i in 0..#10 {
+   writeln(cArray[i]);
+  }
+  c_free(cArray);
+
+Variables of type c_ptr can be compared against or set to nil.
+
+
+Working with strings
+--------------------
+
+If you need to call a C function and provide a Chapel string, you may need to
+convert the Chapel string to a C string first.  Chapel string literals will
+automatically convert to C strings.  A Chapel string variable can be converted
+using the :proc:`string.c_str` method.
+
+myprint.h:
+
+.. code-block:: c
+
+  void myprint(const char* str);
+
+myprint.c:
+
+.. code-block:: c
+
+  void myprint(const char* str) {
+    printf("%s\n", str);
+  }
+
+myprint.chpl:
+
+.. code-block:: chapel
+
+  extern proc myprint(str:c_string);
+
+  // string literal is automatically converted to a c_string
+  myprint("hello");
+
+  // a string variable must be converted with .c_str()
+  var s = "goodbye";
+  myprint(s.c_str());
+
+
+Expressing Dependencies
+=======================
+
+Any required C header files, source code files, object files, or
+library files must be provided to the Chapel compiler by one of
+two mechanisms.
+
+ 1) They can be listed at compile-time on the Chapel command line For
+    example, if an external function foo() was defined in foo.h and foo.c,
+    it could be added to the compilation using any of these commands:
+
+    .. code-block:: sh
+
+       chpl foo.h foo.c myProgram.chpl
+       chpl foo.h foo.o myProgram.chpl #if foo.c had already been compiled)
+       chpl foo.h -lfoo myProgram.chpl #if foo.c had been archived in libfoo.a)
+
+    Note that you can use -I and -L arguments for the Chapel compiler
+    to specify include or library paths as with a C compiler.
+
+  2) Alternatively, the required C resources can be listed within the
```

Comment:
I think this is slightly more indented than the previous bullet "1)", is that intentional?